### ISSUE TITLE
Request for a mid/late-game automation friendly squeezer.

### ISSUE THREAD

**Issue by VampireValik** [CyclopsMC/IntegratedDynamics#175]
For early game the low tech manual squeezer seems absolutely fine. But for mid/late game I feel like there needs to be a more automation friendly squeezer. It can be "expensive" to make. For example require Menril Logs combined with Eye of Ender or something. Throw in a good RF per log cost so it will take a little bit more than a couple of the included generators to power it. That should still allow plenty of time for people to get their jumping exercise in while having a goal to work towards in the form of an automatic machine. Plus it would avoid people needing to create overly complex Rube Goldberg machines to automate the manual squeezer.

Speaking of automating the manual squeezer, it can be done but I think it's a little bit insane how much work it took to find a design. It took:

- A tamed wolf, sitting
- Conveyor belts
- A piston
- A fluid reader
- Entity reader
- 2 Redstone writers
- A redstone repeater
- Blocks (I used glass to see what I was doing, any will do)
- Time

Just finding the right mob took awhile. I tried sheep and pigs but they were unreliable at depressing the squeezer. A sitting wolf seemed to always hit the middle.

Here's how it works (this isn't meant to be a tutorial):

- The entity reader checks if the wolf is on top of the squeezer. If the wolf is there then a redstone writer causes the piston to fire pushing the wolf onto the conveyor belt system. The piston fires on a delay (redstone repeater max setting), otherwise it would fire too fast and the wolf would fall inside the mechanism.
- The conveyor belts take the wolf back up into position where it can fall onto the squeezer again. 
- The fluid reader checks if the drying basin has fluid. If fluid is found a signal is sent via the other redstone writer to reset the squeezer. 
- Item I/O is handled by Refined Storage but could easily be replaced with Integrated Tunnels or any other item transport. 

I'm sure there are smaller and less complicated ways of achieveing the same result but none of the simpler designs I tried worked. I can think of equally complicated designs. I can remove the need for the conveyor belt mod and do it all with pistons but then I'd have to work out all the piston timing issues.

**Comment by rubensworks** [comment]
A late-game squeezer alternative is planned.
But it's not high on my priority list, because the whole idea of this mod is automation.

But I'm happy you had fun automating the manual one, that was my goal ;-)

P.S. Remember that you can also use Forestry's Squeezer.

**Comment by VampireValik** [comment]
I'm not sure fun is the word I would use to describe my endeavor. I was more annoyed at how over-engineered everything needed to be for such a trivial task. I did at least learn a little about how Integrated Dynamics works, though, so that is a positive take-away. This was my first time using readers and writers. Previously I had only used Integrated Tunnels for early game automation.

I don't have Forestry installed. I tried it but bees/tree breeding are all of the nope. The only thing I wanted from Forestry was the automated tree farm. Steve's Carts is more than sufficient for burying me in logs so that is what I use for that task. I guess for now I'm going to have to use my wolf prison to automate the squeezer and hope that you introduce an alternative sooner rather than later.

**Comment by BrewingCoder** [comment]
Personally, I'm fine with the early game squeezer. The forestry squeezer is nicely gain, plus using mod tweak or there's no reason why it can't be added as a byproduct of A pulverizer crusher in another mod. Ender IO is also very flexible when it comes to recipes for the sag mill or alloy furnace.  That's actually the route I took with my mod pack.

**Comment by VampireValik** [comment]
Not everyone uses mod packs. Some of us like to cherry pick the mods we enjoy and put them together without having a bunch of extra stuff we don't care about. 

It's great Integrated Dynamics supports the Forestry Squeezer. That doesn't help me, though. I don't enjoy the content Forestry adds so it's not worth it to me to intsall such a large mod just to access one machine. 

Pulverizer support isn't a bad idea but it does fly in the face of the automation focus mentioned earlier.

Ender IO is not 1.11.2 compatible right now so any integration with it does not help out people playing that version of Minecraft.

There is no reason Integrated Dynamics can't be a stand-alone mod offering everything it needs including an automation-friendly squeezer. If it supports machines from other mods, too, fantastic. As a user, what I expect out of a mod is for it to be stand-alone (if it's not obviously dependent on some other mod) and complete out of the box with no configuration required.

**Comment by BrewingCoder** [comment]
Sorry; yes I get the whole 1.11 issue.  I'm just trying to offer up temporary work-arounds until the feature is implemented.

if you're using EnderIO here's some code to drop in your SAGMILLRecipes_User.xml.  Note that by doing this you're removing the ability for the Sag mill to make pulp from logs.

```
<SAGMillRecipes>
  <!-- Disable log to pulp for Forestry. We need to customize Log Entries -->
  <recipeGroup name="Forestry">
    <recipe name="Log" />
  </recipeGroup>
<recipeGroup name="BrewTech10">
    <recipe name="MenrilLog" energyCost="500">
      <input>
          <itemStack modID="integrateddynamics" itemName="menrilLog"/>
          <itemStack modID="integrateddynamics" itemName="crystalizedMenrilChunk" number="1" chance="0.100" />
      </input>
      <output>
          <itemStack modID="integrateddynamics" itemName="crystalizedMenrilBlock"/>
      </output>
    </recipe>
    <recipe name="MenrilChunk" energyCost="500">
        <input>
            <itemStack modID="integrateddynamics" itemName="crystalizedMenrilBlock"/>
        </input>
        <output>
            <itemStack modID="integrateddynamics" itemName="crystalizedMenrilChunk" number="9"/>
            <itemStack modID="integrateddynamics" itemName="crystalizedMenrilChunk" number="2" chance="0.100" />
            <itemStack modID="integrateddynamics" itemName="crystalizedMenrilChunk" number="2" chance="0.010" />
        </output>
    </recipe>
  </recipeGroup>
</SAGMillRecipes>
```

**Comment by BrewingCoder** [comment]
if you're using Minetweaker and Actually Additions you can add the recipe to the crusher using the following code in a minetweaker script:
`mods.actuallyadditions.Crusher.addRecipe(<integrateddynamics:menrilLog>, <integrateddynamics:crystalizedMenrilBlock>, <integrateddynamics:crystalizedMenrilChunk>, 10);`

**Comment by rubensworks** [comment]
Implementation of the _Mechanical Squeezer_ is finished, and will be included in ID 0.10.0!
